File: U1/EXAMEN/Comercializadora_Soap_DotNet_GR01/Comercializadora_Soap_DotNet_GR01/DTOs/RespuestaDTO.cs
using System.Runtime.Serialization;

namespace Comercializadora_Soap_DotNet_GR01.DTOs
{
    [DataContract]
    public class RespuestaDTO
    {
        [DataMember]
        public bool Exito { get; set; }

        [DataMember]
        public string Mensaje { get; set; }

        [DataMember]
        public ProductoDTO Datos { get; set; }
    }
}

File: U2/PROYECTO/Comercializadora_RestFul_Java_GR01/src/main/java/ec/edu/monster/DTOs/RespuestaDTO.java
package ec.edu.monster.DTOs;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

/**
 * DTO genérico para respuestas
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RespuestaDTO {
    private Boolean exito;
    private String mensaje;
    private Object datos;

    public static RespuestaDTO exitoso(String mensaje) {
        return new RespuestaDTO(true, mensaje, null);
    }

    public static RespuestaDTO exitoso(String mensaje, Object datos) {
        return new RespuestaDTO(true, mensaje, datos);
    }

    public static RespuestaDTO error(String mensaje) {
        return new RespuestaDTO(false, mensaje, null);
    }
}
